src/MassTransit/Util/ChannelExecutor.cs

```diff
@@ -159,14 +159,17 @@ async Task ReadFromChannel()
 
                     async Task RunFuture()
                     {
-                        var task = future.Run();
-
-                        await task.ConfigureAwait(false);
-
-                        _limit.Release();
-
-                        lock (_syncLock)
-                            Monitor.PulseAll(_syncLock);
+                        try
```

Comment:
why? futures never throw.

Reply:
method provided to it might throw, or am I missing something?

Reply:
You're missing something. Look lower.

Reply:
damn , you are right, cool then